gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/oauth2/api/JdbcRefreshTokenRepository.java

```diff
@@ -89,21 +90,21 @@ public Completable deleteByUserId(String userId) {
     }
 
     @Override
-    public Completable deleteByDomainIdClientIdAndUserId(String domainId, String clientId, String userId) {
+    public Completable deleteByDomainIdClientIdAndUserId(String domainId, String clientId, UserId userId) {
         LOGGER.debug("deleteByDomainIdClientIdAndUserId({},{},{})", domainId, clientId, userId);
         return monoToCompletable(getTemplate().delete(JdbcRefreshToken.class)
-                .matching(Query.query(where(SUBJECT).is(userId)
+                .matching(Query.query(where(SUBJECT).is(userId.lookupSubject())
                         .and(where("domain").is(domainId))
                         .and(where("client").is(clientId)))).all())
                 .doOnError(error -> LOGGER.error("Unable to delete refresh token with domain {}, client {} and subject {}",
                         domainId, clientId, userId, error));
     }
 
     @Override
-    public Completable deleteByDomainIdAndUserId(String domainId, String userId) {
+    public Completable deleteByDomainIdAndUserId(String domainId, UserId userId) {
         LOGGER.debug("deleteByDomainIdAndUserId({},{})", domainId, userId);
         return monoToCompletable(getTemplate().delete(JdbcRefreshToken.class)
-                .matching(Query.query(where(SUBJECT).is(userId)
+                .matching(Query.query(where(SUBJECT).is( userId.lookupSubject())
```

Comment:
**question/issue**: I'm supprised that you have to do this lookUp as the tokens (access or refresh are supposed to have the userId as reference) :thinking: